### PR TITLE
Admin loop extra data

### PR DESCRIPTION
**Pull request by sourabhsingh29**

### CODE REVIEW

File: loop/api_admin.py [hunk 1]
```diff
@@ -0,0 +1,1407 @@
...
+        districts  = AdminUser.objects.get(user_id=bundle.request.user.id).get_districts()
+        mandis_list = {}
+        mandi_list[self.mandi_field] = Mandi.objects.filter(district__in=districts)
+        obj = object_list.filter(**mandis_list).distinct()
+        if obj:
+            return True
+        else:
+            raise NotFound("Not allowed to download Mandi")
+
+            # def read_list(self, object_list, bundle):
```
Comment (lokeshgarg92) [inline loop/api_admin.py]: remove extra commented code

File: loop/api_admin.py [hunk 2]
```diff
@@ -0,0 +1,1407 @@
...
+
+    def dehydrate(self, bundle):
+        bundle.data['online_id'] = bundle.data['id']
+        return bundle
+
+
+class DistrictResource(BaseResource):
+    state = fields.ForeignKey(StateResource, 'state', full=True)
+
+    class Meta:
```
Comment (lokeshgarg92) [inline loop/api_admin.py]: add limit parameter

File: loop/api_admin.py [hunk 3]
```diff
@@ -0,0 +1,1407 @@
...
+    hydrate_state = partial(dict_to_foreign_uri, field_name='state')
+
+    def dehydrate(self, bundle):
+        bundle.data['online_id'] = bundle.data['id']
+        return bundle
+
+class BlockResource(BaseResource):
+    district = fields.ForeignKey(DistrictResource, 'district', full=True)
+
+    class Meta:
```
Comment (lokeshgarg92) [inline loop/api_admin.py]: add limit parameter

File: loop/api_admin.py [hunk 4]
```diff
@@ -0,0 +1,1407 @@
...
+
+    def obj_update(self, bundle, request=None, **kwargs):
+        try:
+            bundle = super(VillageResource, self).obj_update(
+                bundle, **kwargs)
+        except Exception, e:
+            block_id = bundle.data['block']['online_id']
+            block = Block.objects.get(id=block_id)
+            attempt = Village.objects.filter(village_name=bundle.data['village_name'],block=block)
+            send_duplicate_message(int(attempt[0].id))
```
Comment (lokeshgarg92) [inline loop/api_admin.py]: put a check on attempt size

File: loop/api_admin.py [hunk 5]
```diff
@@ -0,0 +1,1407 @@
...
+        resource_name = 'loopuser'
+        allowed_methods=['get','put','post']
+        authorization = LoopUserAuthorization('id__in')
+        authentication = ApiKeyAuthentication()
+        always_return_data = True
+
+    hydrate_user = partial(dict_to_foreign_uri, field_name='user')
+    hydrate_village = partial(dict_to_foreign_uri, field_name='village')
+    hydrate_preferred_language = partial(dict_to_foreign_uri,field_name='preferred_language')
+    # hydrate_assigned_villages = partial(dict_to_foreign_uri_m2m, field_name='assigned_villages',
```
Comment (lokeshgarg92) [inline loop/api_admin.py]: remove commented code

File: loop/api_admin.py [hunk 6]
```diff
@@ -0,0 +1,1407 @@
...
+            bundle = super(CropResource, self).obj_create(bundle, **kwargs)
+            # CropLanguage(language=language,crop=bundle.obj,crop_name=crop_name_reg).save()
+        else:
+            send_duplicate_message(int(attempt[0].id))
+        return bundle
+
+    def obj_update(self, bundle, request=None, **kwargs):
+        try:
+            bundle = super(CropResource, self).obj_update(bundle, **kwargs)
+            # cropLanguage = CropLanguage.objects.filter(language=language,crop_id=bundle.data['online_id'])
```
Comment (lokeshgarg92) [inline loop/api_admin.py]: remove commented code

File: loop/api_admin.py [hunk 7]
```diff
@@ -0,0 +1,1407 @@
...
+        else:
+            send_duplicate_message(int(attempt[0].id))
+        return bundle
+    def obj_update(self, bundle, request=None, **kwargs):
+        user = AdminUser.objects.get(user_id=bundle.request.user.id)
+        preferred_language = user.preferred_language.id
+        bundle.data['language'] = "/loop/api/v1/language/%s/" % preferred_language
+        try:
+            bundle = super(CropLanguageResource, self).obj_update(bundle, **kwargs)
+            # cropLanguage = CropLanguage.objects.filter(language=language,crop_id=bundle.data['online_id'])
```
Comment (lokeshgarg92) [inline loop/api_admin.py]: remove commented code

File: loop/api_admin.py [hunk 8]
```diff
@@ -0,0 +1,1407 @@
...
+        else:
+            send_duplicate_message(int(attempt[0].id))
+        return bundle
+    def obj_update(self, bundle, request=None, **kwargs):
+        user = AdminUser.objects.get(user_id=bundle.request.user.id)
+        preferred_language = user.preferred_language.id
+        bundle.data['language'] = "/loop/api/v1/language/%s/" % preferred_language
+        try:
+            bundle = super(VehicleLanguageResource, self).obj_update(bundle, **kwargs)
+            # cropLanguage = CropLanguage.objects.filter(language=language,crop_id=bundle.data['online_id'])
```
Comment (lokeshgarg92) [inline loop/api_admin.py]: remove

File: loop/models.py [hunk 9]
```diff
@@ -88,7 +90,7 @@ class Block(LoopModel):
     block_name = models.CharField(max_length=50)
     district = models.ForeignKey(District)
     is_visible = models.BooleanField(default=True)
-    block_name_en = models.CharField(max_length=100, null=True)
+    block_name_en = models.CharField(max_length=100)
```
Comment (lokeshgarg92) [inline loop/models.py]: Block post save to be written adn write corresponding log handling

File: loop/utils/send_log/loop_admin_log.py [hunk 10]
```diff
@@ -0,0 +1,556 @@
...
+
+def save_admin_loopuser_mandi_child_log(instance,kwargs):
+    AdminUser = get_model('loop','AdminUser')
+    Mandi = get_model('loop','Mandi')
+    Gaddidar = get_model('loop','Gaddidar')
+    GaddidarCommission = get_model('loop','GaddidarCommission')
+    District = get_model('loop','District')
+    LoopUser = get_model('loop','LoopUser')
+    LoopUserAssignedMandi = get_model('loop','LoopUserAssignedMandi')
+    #import pdb;pdb.set_trace()
```
Comment (lokeshgarg92) [inline loop/utils/send_log/loop_admin_log.py]: remove line

File: loop_ivr/migrations/0001_initial.py [hunk 11]
```diff
@@ -8,7 +8,6 @@
 class Migration(migrations.Migration):
 
     dependencies = [
```
Comment (lokeshgarg92) [inline loop_ivr/migrations/0001_initial.py]: discard these changes

File: loop/models.py (hunk 9 above)
Comment (sourabhsingh29) [reply loop/models.py]: resolved

**Pull request merged**